src/impl/tpls/KokkosBlas1_iamax_tpl_spec_decl.hpp

```diff
@@ -699,4 +702,420 @@ KOKKOSBLAS1_CIAMAX_TPL_SPEC_DECL_CUBLAS(unsigned int, Kokkos::LayoutLeft,
+      }                                                                        \
+      Kokkos::Profiling::popRegion();                                          \
+    }                                                                          \
+  };                                                                           \
+  template <class ExecSpace>                                                   \
+  struct Iamax<                                                                \
+      Kokkos::View<INDEX_TYPE, LAYOUT, Kokkos::Device<ExecSpace, MEMSPACE>,    \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
+      Kokkos::View<const double*, LAYOUT, Kokkos::Device<ExecSpace, MEMSPACE>, \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,                  \
+      1, true, ETI_SPEC_AVAIL> {                                               \
+    typedef Kokkos::View<INDEX_TYPE, LAYOUT,                                   \
+                         Kokkos::Device<ExecSpace, MEMSPACE>,                  \
+                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >             \
+        RV;                                                                    \
+    typedef Kokkos::View<const double*, LAYOUT,                                \
+                         Kokkos::Device<ExecSpace, MEMSPACE>,                  \
+                         Kokkos::MemoryTraits<Kokkos::Unmanaged> >             \
+        XV;                                                                    \
+    typedef typename XV::size_type size_type;                                  \
+                                                                               \
+    static void iamax(RV& R, const XV& X) {                                    \
+      Kokkos::Profiling::pushRegion("KokkosBlas::iamax[TPL_ROCBLAS,double]");  \
+      const size_type numElems = X.extent(0);                                  \
+      if (numElems == 0) {                                                     \
+        Kokkos::deep_copy(R, 0);                                               \
+        return;                                                                \
+      }                                                                        \
+      if (numElems < static_cast<size_type>(INT_MAX)) {                        \
+        iamax_print_specialization<RV, XV>();                                  \
+        const int N   = static_cast<int>(numElems);                            \
+        const int XST = X.stride(0);                                           \
+        const int LDX = (XST == 0) ? 1 : XST;                                  \
+        KokkosBlas::Impl::RocBlasSingleton& s =                                \
+            KokkosBlas::Impl::RocBlasSingleton::singleton();                   \
+        KOKKOS_ROCBLAS_SAFE_CALL_IMPL(                                         \
```

Comment:
We should check the pointer mode and only change it if needed.
Also if the initial pointer mode was already `mode_device`, we are currently setting it differently when we exit which will likely confuse users.

Reply:
@lucbv Thanks for the suggestion. I made the changes for the rocBLAS wrappers and for the cuBLAS wrappers as well.